### PR TITLE
STORY-13550 - Upgrade `exception_handling` to support  Rails@7 features

### PR DESCRIPTION
**Pull request by invocandg**
- Added support for ActionMailer 7.x
- Added support for ActionPack 7.x
- Added support for ActiveSupport 7.x

### CODE REVIEW

File: Gemfile [hunk 1]
```diff
@@ -9,14 +9,12 @@ end
...
 
-gem 'actionmailer', '< 6.1'
-gem 'activesupport', '< 6.1'
 gem 'appraisal', '~> 2.2'
 gem 'honeybadger', '~> 4.11'
 gem 'pry'
 gem 'pry-byebug'
 gem 'rake'
-gem 'rspec'
+gem 'rspec', '~> 3.10.0'
```
Comment (retroGiant89) [inline Gemfile:17]: This seems really narrow, do we need to limit it?

Comment (invocandg) [reply Gemfile:17]: Yes, there are some bugs for the versions above, both 3.11.x and 3.12.x.

**Review by retroGiant89 — APPROVED** [review]

File: CHANGELOG.md [hunk 2]
```diff
@@ -2,7 +2,13 @@
 
 Inspired by [Keep a Changelog](https://keepachangelog.com/en/1.0.0/).
 
-Note: this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+**Note:** this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
+
+## [2.15.0] - 2023-03-06
```
Comment (ColinDKelley) [inline CHANGELOG.md:7]: Nit: we usually put "Unreleased" here and then put the date in later, when we release it. Not that it matters much...but it helps us remember that we need to fill the date in.

**Review by ColinDKelley — APPROVED** [review]
Looks good!

File: exception_handling.gemspec [hunk 3]
```diff
@@ -20,9 +20,9 @@ Gem::Specification.new do |spec|
     "allowed_push_host" => "https://rubygems.org"
   }
 
-  spec.add_dependency 'actionmailer',      '>= 5.2', '< 7.0'
-  spec.add_dependency 'actionpack',        '>= 5.2', '< 7.0'
-  spec.add_dependency 'activesupport',     '>= 5.2', '< 7.0'
+  spec.add_dependency 'actionmailer',      '>= 5.2'
+  spec.add_dependency 'actionpack',        '>= 5.2'
+  spec.add_dependency 'activesupport',     '>= 5.2'
```
[the remark this answers is not in the document]
Comment (jebentier) [reply exception_handling.gemspec:25]: Yea I think that’s fine here to be less restrictive so we can upgrade more easily in the future.

**Review by jebentier — APPROVED** [review]

**Review by ColinDKelley — APPROVED** [review]
Looks good!

**Pull request merged**